src/java_bytecode/java_object_factory.cpp

```diff
@@ -1040,8 +1040,10 @@ void java_object_factoryt::gen_nondet_struct_init(
       code.add_source_location()=loc;
       assignments.copy_to_operands(code);
     }
-    else if(name=="@lock")
+    else if(struct_tag == "java.lang.Object")
```

Comment:
This seems odd..? We don't want `@class_identifier` to be set to zero, does JLO have any other fields that `@lock`?